tools/c7n_azure/c7n_azure/functionapp_utils.py

```diff
@@ -116,6 +117,10 @@ def get_function_name(policy_name, suffix):
         function_app_name = policy_name + '-' + suffix
         return re.sub('[^A-Za-z0-9\\-]', '-', function_app_name)
 
+    @staticmethod
+    def _temporary_opener(name, flag, mode=0o777):
```

Comment:
I think we can change it to `get_opener()` function and move to `utils`.

When this is done, here and in functionapp_package we can just do `open(path, 'rb', opener=utils.get_opener())` and avoid code duplication.

Reply:
Great suggestion and will do.